### PR TITLE
(DOCSP-43804)(DOCSP-43803)(DOCSP-43802) BI connector release notes for 2.14.14, 2.14.15, and 2.14.16

### PR DESCRIPTION
**Pull request by sarahsimpers**
[DOCSP 1](https://jira.mongodb.org/browse/DOCSP-43804)
[DOCSP 2](https://jira.mongodb.org/browse/DOCSP-43803)
[DOCSP 3](https://jira.mongodb.org/browse/DOCSP-43802)
[Staging]()

### CODE REVIEW

File: source/release-notes.txt [hunk 1]
```diff
@@ -20,6 +20,67 @@ Release Notes for |bi|
...
+  - macOS 13 arm/OpenSSL 3.0
+  - Ubuntu 18.04 container
+
+- |bi| now supports the upcoming MongoDB version, 8.0.
+- Starting in MongoDB version 8.0, you can convert ``binData`` to
+  ``string`` and ``string`` to ``binData`` by using the :manual:`$convert </reference/operator/aggregation/convert/>`
+  aggregation and providing the required option to specify the format.
+  
+  Converting ``binData`` to ``string`` and ``string`` to ``binData``
+  is unsupported for MongoDB versions 7.0 and earlier, and returns ``null``.
```
Comment (nbagnard) [inline source/release-notes.txt:49]: Now that I see it, I think this is too much details for the release notes.
There are no user facing changes so let's just not get into the details and remove this section.
```suggestion
```

**Review by nbagnard — CHANGES_REQUESTED** [review]

File: source/release-notes.txt (hunk 1 above)
Comment (sarahsimpers) [reply source/release-notes.txt]: @nbagnard Sorry for the delay, I removed this, please take another look when you have a moment and I can merge these in. Thanks!

**Comment by sarahsimpers** [comment]
Rebased to resolve merge conflicts because other release notes were added while this PR was in progress

**Review by nbagnard — APPROVED** [review]
LGTM!

**Pull request merged**